src/middleware/externalRequests.js

```diff
@@ -63,8 +69,284 @@ const prepareRequestConfig = requestDetails => {
 exports.requestsMiddleware = () => async (ctx, next) => {
   await prepareLookupRequests(ctx)
   await next()
+  await orchestrateMappingResult(ctx)
+}
+
+// For now only json data is processed
+const orchestrateMappingResult = async ctx => {
```

Comment:
This function is quite massive and a bit hard to read when scrolling down. Lets move some of this logic into its own function to make it easier to test and to see what the main function is actually doing

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/36/commits/0ad1c733cedba9bce10ea19225ccfd2491899392